.github/workflows/auto_author_assign.yml

```diff
@@ -0,0 +1,15 @@
+# https://github.com/marketplace/actions/auto-author-assign
+name: 'Auto Author Assign'
+
+on:
+  pull_request_target:
+    types: [opened, reopened]
+
+permissions:
+  pull-requests: write
+
+jobs:
+  assign-author:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: toshimaru/auto-author-assign@v1.3.4
```

Comment:
I would like to gently push back on this on two fronts:
- I am getting e-mail fatigue from binder/assignment/welcome bot etc; do we really want to send five extra e-mails to every new contributor (open issue: 1. welcome, open PR: 2. thanks for opening PR, 3. binder, 4. assign, 5. congrats on first merged PR)? For those who subscribe this becomes just a noise and may start ignoring e-mails which actually contain important info.
- this workflow is from a third-party; in light of codecov issue earlier this year (see https://github.com/jupyterlab/team-compass/issues/125 and https://github.com/jupyterhub/team-compass/issues/404, https://github.com/jupyterhub/team-compass/issues/398) it might be preferable to minimize use of actions which do not come from trusted sources.

Reply:
I gently disagree. This PR is part of a larger effort to add tools that allow the project to be more easily managed and maintained.

>I am getting e-mail fatigue from binder/assignment/welcome bot etc;

I have not used email notifications from Github for years, and I would suggest disabling them. After contributing to some projects it stops being useful. If you still prefer to use them then I would advice to make filters on the inbox so you can skip the inbox and add them to specific labels.

>(open issue: 1. welcome, open PR: 2. thanks for opening PR, 3. binder, 4. assign, 5. congrats on first merged PR)?

4 already sounds really bad. But I do not think for a first time contributor it will be an issue.

>For those who subscribe this becomes just a noise and may start ignoring e-mails which actually contain important info.

The reason for this initisal work is trying to automate workflows and making it easier for folks to know what they are working on, via tools like zenhub (more on that on the next JLab dev meeting). 

>this workflow is from a third-party; in light of codecov issue earlier this year (see jupyterlab/team-compass#125 and jupyterhub/team-compass#404, jupyterhub/team-compass#398) it might be preferable to minimize use of actions which do not come from trusted sources.

The labeler was originally from a third party and now is back to using an action from github itself. The auto assign workflow has minimal code that can be audited if we really want to go in that direction.



Here it is:

https://github.com/toshimaru/auto-author-assign/blob/main/src/index.js

<img width="757" alt="Screen Shot 2021-06-01 at 8 22 26 AM" src="https://user-images.githubusercontent.com/3627835/120330519-81a56700-c2b2-11eb-8c57-cc7ebc2f6a16.png">